packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -223,47 +224,57 @@ impl Abigen {
                 all_properties.push(prop);
             }
         }
+        all_properties
+    }
+
+    /// Reads the parsed ABI and returns the custom types (either `struct` or `enum`) found in it.
+    fn get_custom_types(abi: &JsonABI) -> HashMap<String, Property> {
+        let mut custom_types = HashMap::new();
+        let mut nested_custom_types: Vec<Property> = Vec::new();
+
+        let all_properties = Abigen::get_all_properties(abi);
 
         for prop in all_properties {
-            if prop.type_field.contains(type_string) {
-                // Top level struct
-                let custom_type_name = extract_custom_type_name_from_abi_property(prop, ty)
+            if is_custom_type(prop) {
+                // Top level custom type
+                let custom_type_name = extract_custom_type_name_from_abi_property(prop, None)
                     .expect("failed to extract custom type name");
-                structs
+                custom_types
                     .entry(custom_type_name)
                     .or_insert_with(|| prop.clone());
 
-                // Find inner structs in case of nested custom types
+                // Find inner {structs, enums} in case of nested custom types
                 for inner_component in prop.components.as_ref().unwrap() {
-                    inner_structs.extend(Abigen::get_inner_custom_properties(
-                        inner_component,
-                        type_string,
-                    ));
+                    nested_custom_types
+                        .extend(Abigen::get_nested_custom_properties(inner_component));
                 }
             }
         }
 
-        for inner_struct in inner_structs {
-            let inner_custom_type_name =
-                extract_custom_type_name_from_abi_property(&inner_struct, ty)
-                    .expect("failed to extract custom type name");
-            structs
-                .entry(inner_custom_type_name)
-                .or_insert(inner_struct);
+        for nested_custom_type in nested_custom_types {
+            if is_custom_type(&nested_custom_type) {
```

Comment:
Isn't this check redundant? We're adding the custom types to `nested_custom_types` by using `get_nested_custom_properties()` [0] which already checks for `is_custom_type()` [1].

[0]:

```rust
// Find inner {structs, enums} in case of nested custom types
for inner_component in prop.components.as_ref().unwrap() {
    nested_custom_types
        .extend(Abigen::get_nested_custom_properties(inner_component));
}
```

[1]:

```rust
 fn get_nested_custom_properties(prop: &Property) -> Vec<Property> {
    let mut props = Vec::new();

    if is_custom_type(prop) {
        props.push(prop.clone());

        for inner_prop in prop.components.as_ref().unwrap() {
            let inner = Abigen::get_nested_custom_properties(inner_prop);
            props.extend(inner);
        }
    }

    props
}
```

Reply:
Yup, good catch